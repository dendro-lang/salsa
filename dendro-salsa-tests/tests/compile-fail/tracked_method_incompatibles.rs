#[salsa::jar(db = Db)]
struct Jar(Tracked);


#[salsa::tracked(jar = Jar)]
struct Tracked {
    field: u32,    
}


impl Tracked {
    #[salsa::tracked]
    fn use_tracked(&self) {

    }
}

trait Db: salsa::DbWithJar<Jar> {}


fn main() {}