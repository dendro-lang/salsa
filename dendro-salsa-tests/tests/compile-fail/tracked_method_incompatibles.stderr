error: #[salsa::tracked] must also be applied to the impl block for tracked methods
  --> tests/compile-fail/tracked_method_incompatibles.rs:13:20
   |
13 |     fn use_tracked(&self) {
   |                    ^
